VAGOV-1637 Health Care Region Locations list page

aurora-a-k-a-lightning opened the pull request:
## Description
- Add the locations list for a health care region grouped by "main location" sorted by name alphabetically

## Testing done
- Locally with staging data

## Screenshots
<img width="523" alt="screen shot 2019-02-21 at 11 29 17 am" src="https://user-images.githubusercontent.com/19178435/53196079-f5347600-35cb-11e9-8ef0-6120e199c08c.png">

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

beeyayjay committed:
Add templates and query for facility landing pages from drupal

beeyayjay committed:
[WIP] Templates and query for facility region pages from drupal

aurora-a-k-a-lightning committed:
WIP facilities page for region

aurora-a-k-a-lightning committed:
WIP facilities page for region - listed facilties

aurora-a-k-a-lightning committed:
style locations page

aurora-a-k-a-lightning committed:
Merge remote-tracking branch 'origin/VAGOV-1627-facility-landing-page-drupal' into vagov-health-care-region

aurora-a-k-a-lightning committed:
Merge branch 'vagov-health-care-region' into vagov-1637-2

aurora-a-k-a-lightning committed:
add facility pictures to locations page

aurora-a-k-a-lightning committed:
extract method to create health care region list pages

aurora-a-k-a-lightning committed:
Merge branch 'master' of https://github.com/department-of-veterans-affairs/vets-website into vagov-1637-3

aurora-a-k-a-lightning committed:
merge with master

jbalboni commented on src/site/stages/build/drupal/metalsmith-drupal.js:
```diff
@@ -42,6 +42,29 @@ function writeDrupalIndexPage(files) {
+// Creates the top-level health care region list pages (Locations, Services, etc.)
+function createHealthCareRegionListPages(page, drupalPagePath, files) {
+  // Create the top-level locations page for Health Care Regions
+  files[`drupal${drupalPagePath}/locations/index.html`] = createFileObj(
```
What's the reason behind adding `locations` to the path here? I'm a little hesitant to have url info embedded in our build scripts, instead of being sourced from Drupal.

ahay-agile6 replied:
i was trying to ask this in slack but i don't think the message went through! these listing pages for health care regions (Locations, health services, etc.) are reverse fields on the health care region page node, so, they do not have nodes themselves, nor do they have urls. only the detail pages do.
how should we proceed?

jbalboni replied:
I think this is ok, then. I'll think about maybe finding a better place to put this (especially if we end up having more like this), but this is fine for now.

jbalboni commented on src/applications/static-pages/facilities/FacilityListWidget.jsx:
```diff
@@ -0,0 +1,156 @@
+      return 0;
+    });
+
+    return sortedFacilities.map(facility => {
```
I think I would return `sortedFacilities` from this function and move this map into the jsx in the render method. It's a little bit easier to read if the primary jsx output is in the render method of a component.

jbalboni commented on src/applications/static-pages/facilities/FacilityListWidget.jsx:
```diff
@@ -0,0 +1,156 @@
+              </div>
+            </div>
+            <div className="location-details-link">
+              <a href="#" className="usa-button usa-button-secondary">
```
Is this going to have a link eventually?

ahay-agile6 replied:
yes, once we get the facility details pages

jbalboni commented on src/applications/static-pages/facilities/FacilityListWidget.jsx:
```diff
@@ -0,0 +1,156 @@
+    super(props);
+    this.state = {
+      loading: true,
+      parsedFacilities: JSON.parse(this.props.facilities),
```
I think I would parse this in the createFacilityListWidget file, before passing it into your React component. That creates a cleaner separation between the code that's dealing with the DOM and data stored in it, and your React code.

It'll also make it easier to unit test this component.

jbalboni reviewed (commented):
I left a few comments, but this looks pretty good! We also usually write unit tests for components like FacilityListWidget, which you can dive into in this ticket or in a future one.

aurora-a-k-a-lightning committed:
code review updates

- parse facilities json before sending to react widget
- refactor mapping of facilities to be easier to read

aurora-a-k-a-lightning committed:
Merge branch 'master' of https://github.com/department-of-veterans-affairs/vets-website into vagov-1637-3

aurora-a-k-a-lightning committed:
facility list widget: refactor and add tests

jbalboni committed:
Set promise to instance variable

jbalboni commented on src/platform/utilities/environment/index.js:
```diff
@@ -35,7 +35,7 @@ const ENVIRONMENT_CONFIGURATIONS = {
     BASE_URL: `http://${location.hostname}${
       location.port ? `:${location.port}` : ''
     }`,
-    API_URL: `http://${location.hostname}:3000`,
```
Need to revert this change

jbalboni commented on src/site/stages/build/drupal/graphql/GetAllPages.graphql.js:
```diff
@@ -11,12 +13,14 @@ module.exports = `
   ${fragments}
   ${landingPage}
   ${page}
+  ${healthCareRegionPage}
```
I think this needs to go in the other query files, too.

aurora-a-k-a-lightning committed:
merge with master

aurora-a-k-a-lightning committed:
Merge branch 'vagov-1637-3' of https://github.com/department-of-veterans-affairs/vets-website into vagov-1637-3

jbalboni committed:
Skip lifecycle method

aurora-a-k-a-lightning committed:
Update GetAllPages.graphql.js

aurora-a-k-a-lightning committed:
Merge branch 'vagov-1637-3' of https://github.com/department-of-veterans-affairs/vets-website into vagov-1637-3

aurora-a-k-a-lightning committed:
revert environment api endpoint change
